.circleci/config.yml

```diff
@@ -0,0 +1,251 @@
+aliases:
+  - &restore-cache
+    keys:
+      - v1-dependencies-{{ .Branch }}-{{ checksum "package.json" }}
+      # Fallback in case checksum fails
+      - v1-dependencies-{{ .Branch }}-
+  - &save-cache
+    paths:
+      - node_modules
+    key: v1-dependencies-{{ .Branch }}-{{ checksum "package.json" }}
+
+  - &restore-cache-website
+    keys:
+      - v1-website-dependencies-{{ .Branch }}-{{ checksum "website/package.json" }}
+      # Fallback in case checksum fails
+      - v1-website-dependencies-{{ .Branch }}-
+  - &save-cache-website
+    paths:
+      - website/node_modules
+    key: v1-website-dependencies-{{ .Branch }}-{{ checksum "website/package.json" }}
+
+  - &restore-cache-danger
+    keys:
+      - v1-danger-dependencies-{{ .Branch }}-{{ checksum "danger/package.json" }}
+      # Fallback in case checksum fails
+      - v1-danger-dependencies-{{ .Branch }}-
+  - &save-cache-danger
+    paths:
+      - danger/node_modules
+    key: v1-danger-dependencies-{{ .Branch }}-{{ checksum "danger/package.json" }}
+
+  - &restore-cache-android
+    keys:
+      - v1-android-dependencies-{{ .Branch }}-{{ checksum "ReactAndroid/build.gradle" }}
+      # Fallback in case checksum fails
+      - v1-android-dependencies-{{ .Branch }}-
+  - &save-cache-android
+    paths:
+     - "ReactAndroid/build/downloads"
+    key: v1-android-dependencies-{{ .Branch }}-{{ checksum "ReactAndroid/build.gradle" }}
+
+  - &restore-cache-buck
+    keys:
+# TODO: Pick file to checksum against
```

Comment:
I just realized with CircleCI caching we'd stop using the latest copy of Buck. If we want that, one thing we could do is commit a small file to RN that contains just a Git commit hash or Git tag of the Buck repo. We'd check out just that commit and cache it. That way we intentionally don't use the latest copy of Buck and also consistently use the same version, not at the whim of CircleCI's caching behavior.

If we do want to use the latest copy of Buck, we could create a shallow clone of its Git repo (to avoid downloading 500MB) and build it each time.